akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -1140,8 +1153,8 @@ private[cluster] class OnMemberUpListener(callback: Runnable) extends Actor with
     }
   }
 
-  def isSelfUp(m: Member): Boolean =
-    m.uniqueAddress == cluster.selfUniqueAddress && m.status == MemberStatus.Up
+  private def checkChanged(m: Member, previousStatus: MemberStatus = null): Boolean =
+    m.uniqueAddress == cluster.selfUniqueAddress && (if (prefer == Down) previousStatus == prefer else m.status == prefer)
```

Comment:
In fact I think it would be nice to make the 
```down(address:Address)```
```leave(address:Address)```
```join(address:Address)```
to return a ```Future```
or add method like ```leave(address:Address,timeout:FinitTimeOut)``` which return a Future,so I add that here.
wdyt?